sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -482,6 +491,13 @@ def query_entities(
                 :dedent: 8
                 :caption: Query entities held within a table
         """
+        entity_hook = kwargs.pop("entity_hook", None)
+        page_iterator_class = TableEntityPropertiesPaged
+        if entity_hook:
+            page_iterator_class = functools.partial(
+                TableEntityPropertiesPaged, entity_hook=entity_hook
+            )
```

Comment:
I don't think you should need to do this - I'm pretty sure any "extra" kwargs you pass into the `ItemPaged` constructor below will automatically be passed into the `TableEntityPropertiesPaged` constructor.